### PR TITLE
Fix Slash Commands

### PR DESCRIPTION
**Pull request by daniel-panhead**
# TL;DR

Fixed all existing slash commands except `/start-verification`, `/add-members`, and `/check-email` + migrated `role-selector` command to slash command.

All commands with persistent embeds/messages now also save info to Firebase so that they can continue to be used if the bot is restarted.

# Description

- `/clear-chat` - fix keep_pinned option
- `/self-care` - fix self-care tip loading and message posting
- `/start-report` - fix message sending
- `/discord-contests` - fix question loading and posting
- `/load-question` - fix question saving
- `/start-mentor-cave` - add message ID saving functionality
- `/role-selector` - converted from prefix command + add message ID saving functionality

### New collection: `SavedMessages`

- New doc: 'mentor-cave'

```
{
    adminControls: { // stores info for admin control message
        channelId: string,
        messageId: string
    },
    extraEmojis: {
        [emojiIcon: string]: string // icon mapped to actual display name
    },
    requestTicketConsole: { // stores info for dropdown ticket request message
        channelId: string,
        messageId: string,
        publicRoleId: string,
        reminderTime: number
    },
    roleReactions: { // stores info for role reaction message
        channelId: string,
        messageId: string
    },
    ticketCount: number
}
```

- New doc: 'pronouns'

```
{ // stores info for pronoun reaction message
    channelId: string,
    messageId: string
}
```

- New doc: `role-selector`

```
{
    emojis: {
        [emojiIcon: string]: { // icon mapped to role info
            description: string,
            title: string,
            roleId: string,
        }
    }
    // Message info for reaction selector embed message
    channelId: string,
    messageId: string
}
```

### New Collection: `Questions`

- Each question doc (autogen ID) has the following schema:

```
{
    answers: string[] | number[], // array containing possible answers - set to empty array for manual graded answer
    asked: boolean,
    needAllAnswers: boolean,
    question: string
}
```

Questions can be loaded with the `/load-question` command where a json doc containing an array of objects with the above schema is uploaded.

### New Collection: `QuestionsLeaderboard`

- Each doc (autogen ID) represents a member on the leaderboard:

```
{
    memberId: string,
    points: number
}
```

### New Collection: `Reminders`

- Each doc (autogen ID) has the following schema:

```
{
    reminder: string,
    sent: boolean
}
```

## Why

This allows logs to actually use critical Discord bot commands + more easily recover if the bot crashes (no more rerunning all the commands)

## User Changes

No changes to the user experience - things work now :)

## Tests Done

For each command, the functionality was tested + if containing persistent info, it was also tested that functionality continued to work after restarting the bot.

# Breaking Changes

Commands were converted from referencing MongoDB info to referencing Firebase info.

### CODE REVIEW

**Comment by byronwang93** [comment]
DANG you're cracked i shall work on reviewing this today

**Review by byronwang93 — APPROVED** [review]
cracked
